codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -199,11 +204,23 @@ exception UnexpectedRegisterDoesNotExist(id)
 Updates a given register with new latest block values and new fetched
 events.
 */
-let updateRegister = (self: t, ~latestFetchedBlock, ~newFetchedEvents, ~isFetchingAtHead) => {
-  ...self,
-  isFetchingAtHead,
-  latestFetchedBlock,
-  fetchedEventQueue: List.concat(self.fetchedEventQueue, newFetchedEvents),
+let updateRegister = (
+  self: t,
+  ~latestFetchedBlock,
+  ~newFetchedEvents: list<Types.eventBatchQueueItem>,
+  ~isFetchingAtHead,
+) => {
+  let firstEventBlockNumber = switch self.firstEventBlockNumber {
+  | Some(n) => Some(n)
+  | None => newFetchedEvents->Utils.List.getLast->Option.map(v => v.blockNumber)
```

Comment:
Why do you get last here instead of the first?

Reply:
Ah good catch sorry I had it backwards in my head!